src/math/exp_fcn.c

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+/* Include Files */
+#include <sof/math/exp_fcn.h>
+#include <sof/math/numbers.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+/* Function Declarations */
+static inline int64_t lomul_s64_SR_sat_near(int64_t a, int64_t b);
+
+static inline void wide_mul_s64(int64_t in_0, int64_t in_1, uint64_t *bitshiptrout,
+				uint64_t *bitsloptrout);
+/*
+ * Arguments	: int64_t a
+ *		  int64_t b
+ * Return Type	: int64_t
+ */
+static inline int64_t lomul_s64_SR_sat_near(int64_t a, int64_t b)
+{
+	uint64_t u64_rhi;
+	uint64_t u64_rlo;
+	bool roundup;
+
+	wide_mul_s64(a, b, &u64_rhi, &u64_rlo);
+	roundup = ((u64_rlo & BIT_MASK_LOW_Q27P5) != 0ULL);
```

Comment:
There's no need to use xULL for every small number. It makes code less pleasant to read. Just compare to 0.